### PR TITLE
Comment archival and viewing

### PR DESCRIPTION
**Pull request by Owez**

### CODE REVIEW

**Comment by Owez** [comment]
This issue might clash with #71 because it adds more to metadata

File: yark/__init__.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
 - `Channel`
     - `DownloadConfig`
 - `Video`
+    - `Element`
```
Comment (Owez) [inline yark/__init__.py:13]: Remove this

File: yark/__init__.py [hunk 2]
```diff
@@ -23,7 +24,7 @@
 """
 
 from .channel import Channel, DownloadConfig
-from .video import Video, Element, Note, Thumbnail
+from .video import Video, Element, Element, Note, Thumbnail
```
Comment (Owez) [inline yark/__init__.py:27]: Remove this

File: yark/video.py [hunk 3]
```diff
@@ -197,14 +203,14 @@ def _magnitude(count: Optional[int] = None) -> str:
 
 
 class Element:
-    video: Video
+    parent: Video | Comment | Channel | CommentAuthor
     inner: dict[datetime, Any]
 
     @staticmethod
-    def new(video: Video, data):
+    def new(parent: Video | Comment | Channel, data):
```
Comment (Owez) [inline yark/video.py:210]: Add `CommentAuthor` here

File: yark/video.py [hunk 4]
```diff
@@ -217,9 +223,18 @@ def update(self, kind: Optional[str], data):
             # Update
             self.inner[datetime.utcnow()] = data
 
-            # Report if wanted
+            # Report if wanted and we have a parent channel
```
Comment (Owez) [inline yark/video.py:226]: Remove, all have parent channels

**Comment by Owez** [comment]
Comment archival now works but isn't tested 😎 
Need to test & polish to figure out edge cases and then it's onto getting comments working in the viewer

**Comment by Owez** [comment]
Comment author icons are almost completed, not sure if i should do `author_icons` or `comment_author_icons` or something.

image

**Comment by Owez** [comment]
I don't think this can be merged until #71 because it adds a decent amount of time to metadata fetching and makes medium-sized channels take ages. Might enable it with a `--comments` flag so people can use it but not by default.

**Comment by Owez** [comment]
I'll add it in and gate it behind `--comments` if #71 if perf is too much of an issue

**Comment by Owez** [comment]
I'll do viewing in a seperate PR later in the month when I'm doing changes on the viewer, for now this is enough.
